src/test/testFrontierDistance.cpp

```diff
@@ -0,0 +1,163 @@
+	unsigned distance1 = milBld1->GetFrontierDistance();
+
+	BOOST_REQUIRE_EQUAL(distance0, 3u);
+	BOOST_REQUIRE_EQUAL(distance0, distance1); 
+}
+
+BOOST_FIXTURE_TEST_CASE(FrontierDistanceMiddle, FrontierWorldMiddle)
+{
+	MapPoint milBld0Pos = MapPoint(1, 20);
+	nobMilitary*  milBld0 = dynamic_cast<nobMilitary*>(BuildingFactory::CreateBuilding(world, BLD_WATCHTOWER, milBld0Pos, 0, NAT_BABYLONIANS));
+
+	MapPoint milBld1Pos = MapPoint(31, 20);
+	nobMilitary*  milBld1 = dynamic_cast<nobMilitary*>(BuildingFactory::CreateBuilding(world, BLD_WATCHTOWER, milBld1Pos, 1, NAT_BABYLONIANS));
+
+	unsigned distance0 = milBld0->GetFrontierDistance();
+	unsigned distance1 = milBld1->GetFrontierDistance();
+
+	BOOST_REQUIRE_EQUAL(distance0, 1u);
+	BOOST_REQUIRE_EQUAL(distance0, distance1);
+}
+
+BOOST_FIXTURE_TEST_CASE(FrontierDistanceInland, FrontierWorldBig)
+{
+	MapPoint milBld0Pos = MapPoint(1, 60);
+	nobMilitary*  milBld0 = dynamic_cast<nobMilitary*>(BuildingFactory::CreateBuilding(world, BLD_WATCHTOWER, milBld0Pos, 0, NAT_BABYLONIANS));
+
+	MapPoint milBld1Pos = MapPoint(61, 60);
+	nobMilitary*  milBld1 = dynamic_cast<nobMilitary*>(BuildingFactory::CreateBuilding(world, BLD_WATCHTOWER, milBld1Pos, 1, NAT_BABYLONIANS));
+
+	unsigned distance0 = milBld0->GetFrontierDistance();
+	unsigned distance1 = milBld1->GetFrontierDistance();
+
+	BOOST_REQUIRE_EQUAL(distance0, 0u);
+	BOOST_REQUIRE_EQUAL(distance0, distance1);
+}
+
+BOOST_FIXTURE_TEST_CASE(FrontierDistanceNearFrontierAddonActive, FrontierWorldSmall)
+{
+	DescIdx<TerrainDesc> tWater(0);
+	for (; tWater.value < world.GetDescription().terrain.size(); tWater.value++)
+	{
+		if (world.GetDescription().get(tWater).kind == TerrainKind::WATER && !world.GetDescription().get(tWater).Is(ETerrain::Walkable))
+			break;
+	}
+	this->ggs.setSelection(AddonId::FRONTIER_DISTANCE_REACHABLE, 1);
+	for (int i = 1; i < 20; i++) {
+		world.GetNodeWriteable(MapPoint(5, i)).t1 = tWater;
+	}
+
+	MapPoint milBld0Pos = MapPoint(1, 10);
+	nobMilitary*  milBld0 = dynamic_cast<nobMilitary*>(BuildingFactory::CreateBuilding(world, BLD_WATCHTOWER, milBld0Pos, 0, NAT_BABYLONIANS));
+
+	MapPoint milBld1Pos = MapPoint(10, 10);
+	nobMilitary*  milBld1 = dynamic_cast<nobMilitary*>(BuildingFactory::CreateBuilding(world, BLD_WATCHTOWER, milBld1Pos, 1, NAT_BABYLONIANS));
+
+	unsigned distance0 = milBld0->GetFrontierDistance();
+	unsigned distance1 = milBld1->GetFrontierDistance();
+
+	BOOST_REQUIRE_EQUAL(distance0, 0u);
+	BOOST_REQUIRE_EQUAL(distance0, distance1);
+}
+
+
+BOOST_FIXTURE_TEST_CASE(FrontierDistanceMiddleAddonActive, FrontierWorldMiddle)
+{
+	DescIdx<TerrainDesc> tLava(0);
+	for (; tLava.value < world.GetDescription().terrain.size(); tLava.value++)
+	{
+		if (world.GetDescription().get(tLava).kind == TerrainKind::LAVA && !world.GetDescription().get(tLava).Is(ETerrain::Walkable))
+			break;
+	}
+
+	this->ggs.setSelection(AddonId::FRONTIER_DISTANCE_REACHABLE,1 );
+	for (int i = 1; i < 20; i++) {
```

Comment:
This is not `world.getHeight()` (50). Is that on purpose?

Reply:
no, just a copy & paste mistake.